src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -5282,19 +5281,48 @@ void wm_vuldet_create_and_send_message(vu_task_ctx_t* task, task_status TASK_STA
     }
 }
 
-void wm_vuldet_on_demand_scan(vu_task_ctx_t* current_task, wm_vuldet_t* vuldet) {
+bool wm_vuldet_on_demand_scan(vu_task_ctx_t* current_task, wm_vuldet_t* vuldet) {
+    bool retry_agent = false;
+    char status_message[OS_SIZE_64] = {0};
+
+    if (!current_task || !vuldet) {
+        return false;
+    }
+
+    if (!current_task->agent_data) {
+        os_calloc(1, sizeof(scan_agent), current_task->agent_data);
+    } else {
+        time_t sleep_time = current_task->agent_last_scan_time + vuldet->retry_interval - time(NULL);
+        if (sleep_time > 0) {
+            mtdebug1(WM_VULNDETECTOR_LOGTAG, "Going to sleep %ld seconds before retrying pending agents", sleep_time);
+            sleep(sleep_time);
```

Comment:
Instead of sleeping wouldn't be better if the working thread continues its work?

Reply:
The whole point of the retry logic is to give **syscollector** time to synchronize the packages that weren't ready.
If after pushing the pending task to the end of the queue the time was spent working, it won't sleep at all.